File: server/data/results/translations/anchored/java/run_length_encode.java
public class Main {
    public static String encode(String text) {
        if (text.length() == 0) {
            return "";
        }
        String result = "";
        char current = text.charAt(0);
        int count = 1;
        int i = 1;
        while (i < text.length()) {
            if (text.charAt(i) == current) {
                count = count + 1;
            } else {
                result = result + current + count;
                current = text.charAt(i);
                count = 1;
            }
            i = i + 1;
        }
        result = result + current + count;
        return result;
    }

    public static void main(String[] args) {
        for (String sample : new String[]{"aaabbc", "abcd", "zzzzzzzz"}) {
            System.out.println(encode(sample));
        }
    }
}

File: server/data/results/translations/anchored/javascript/run_length_encode.js
function encode(text) {
    if (text.length === 0) {
        return "";
    }
    let result = "";
    let current = text[0];
    let count = 1;
    let i = 1;
    while (i < text.length) {
        if (text[i] === current) {
            count = count + 1;
        } else {
            result = result + current + String(count);
            current = text[i];
            count = 1;
        }
        i = i + 1;
    }
    result = result + current + String(count);
    return result;
}


function main() {
    for (const sample of ["aaabbc", "abcd", "zzzzzzzz"]) {
        console.log(encode(sample));
    }
}


main();